Bumps version to 6.0.28 Official Release

gnkidwell opened the pull request:
Bumps version to 6.0.28 Official Release, this is according to Ubiquiti a stable release.

Edit: Since I have deleted many different pulls of the 6.0.2x controller, due to incredibly fast releases by Ubiquiti, I must repeat that Python37 was replaced with Python27 due to V8 having a dependency for 27. Mongodb can also run on Python27, so this is to potentially declutter the script as well as not need an additional pkg installed. png was also added as it is a dependency for freetype2. Also the order of all of the packages was reordered to make sure dependencies are installed before their packages... 

Cannot downgrade from 6.1.x without a full uninstall of the UniFi controller... This did install clean on fresh virtual machines running on:

PFSense 2.4.5
FreeBSD 11.3-STABLE

OPNSense 20.7.3
FreeBSD 12.1-RELEASE

Link to Ubiquiti's forum about the release:

https://community.ui.com/releases/UniFi-Network-Controller-6-0-28/23c77a99-0957-449a-b3c0-58a37d4df81f

Install link:

fetch -o - https://git.io/JTD2a | sh -s

Install link with text output (If you do decide to install it, and feel like sharing your Bash results, please upload the text file)

fetch -o - https://git.io/JTD2a | sh -s cmd |& tee unifi6028.txt

tong2x commented:
i'm getting 
"fetch: https://git.io/JTzmu: No address record"
when trying both link

gnkidwell commented:
> i'm getting
> "fetch: https://git.io/JTzmu: No address record"
> when trying both link
> I'm using OPNSense firewall

I am not sure why you are getting no address record, unless there is a leading space on your shell... I have tried this again on my production box and my virtual machine. I am not having these issues... This sounds like you are having DNS issues. Try this below...

host -a git.io

If you are not resolving that, you are having DNS complications, this sounds like an issue you just had on the previous build as well... Are you running Suricata or Snort, if you are you may want to check that git.io is not being blocked by one of your rules. 

![image](https://user-images.githubusercontent.com/71857593/96824425-65744a00-13f4-11eb-9726-ec7e5bff0573.png)

tong2x commented:
ok first error is internet is down on the test box

now im still getting java not found at end of installation wherein the service does not actually run also
tried on 2 seperate machine already, seem link to java is broken

btw Im using opnsense firewall attached is the log file for both machines

manually running the service returns

_**root@OPNsense:~ # service unifi.sh start
Starting UniFi controller.
eval: /usr/local/bin/java: not found**_


[unifi6028.txt](https://github.com/gozoinks/unifi-pfsense/files/5420700/unifi6028.txt)
[unifi6028-I5.txt](https://github.com/gozoinks/unifi-pfsense/files/5420732/unifi6028-I5.txt)

gnkidwell commented:
> ok first error is internet is down on the test box
> 
> now im still getting java not found at end of installation wherein the service does not actually run also
> tried on 2 seperate machine already, seem link to java is broken
> 
> btw Im using opnsense firewall attached is the log file for both machines
> 
> manually running the service returns
> 
> _**root@OPNsense:~ # service unifi.sh start Starting UniFi controller. eval: /usr/local/bin/java: not found**_
> 
> [unifi6028.txt](https://github.com/gozoinks/unifi-pfsense/files/5420700/unifi6028.txt)
> [unifi6028-I5.txt](https://github.com/gozoinks/unifi-pfsense/files/5420732/unifi6028-I5.txt)

So absolutely nothing has changed in the shell script to keep java from running or installing. The UniFi controller also does not keep java from running or installing. These installs according to the text files are not fresh installs but are being done over the same 2 boxes. Have you tried doing this in a fresh vm? I am not sure why your Java is broken, I will fire up OpnSense in a VM today to see if it will install in a fresh vm. Aside from that, this script could not be causing your Java to fail, it is also evident that Java is installed via the text files. Where in the text files are you seeing the Java link is broken?

tong2x commented:
there is no "java" only "javavm"

running service after vi returns

**root@OPNsense:/usr/local/bin # service unifi.sh start
Starting UniFi controller.
eval: /usr/local/bin/java: Permission denied
root@OPNsense:/usr/local/bin #**

gnkidwell commented:
> there is no "java" only "javavm"
> 
> running service after vi returns
> 
> **root@OPNsense:/usr/local/bin # service unifi.sh start Starting UniFi controller. eval: /usr/local/bin/java: Permission denied root@OPNsense:/usr/local/bin #**
> 
> could it be unifi itself? it seems running the service returns/shows the message
> 
> or an update to opnsense, but I'm not aware of any update to opnsense recently

This is not UniFi once again, UniFi uses java to run. Java will need permissions to that file make sure your permissions are the same as below. 
![image](https://user-images.githubusercontent.com/71857593/96879706-fa516480-1441-11eb-91b3-13410e8f3523.png)


I am seeing the current version of OpnSense is 20.7, can you confirm which version you are running?

tong2x commented:
yes correct
OPNsense 20.7.3-amd64
FreeBSD 12.1-RELEASE-p10-HBSD

will check tommorrow for the file permission

gnkidwell commented:
> yes correct
> OPNsense 20.7.3-amd64
> FreeBSD 12.1-RELEASE-p10-HBSD
> 
> will check tommorrow for the file permission

Hey just want to let you know. I got OpnSense installed in a virtual machine, I have confirmed the issue is with your java. Fresh Virtual Machine I just get the standard WARN message, and the UniFi controller is active and functioning. You could always give the script one more go now that we recreated the java file. If that does not work, you will indeed need to fix the permissions on the java file. 

![image](https://user-images.githubusercontent.com/71857593/96896732-7b652780-1453-11eb-96d6-9514d3c4673f.png)

gnkidwell commented:
> yes correct
> OPNsense 20.7.3-amd64
> FreeBSD 12.1-RELEASE-p10-HBSD
> 
> will check tommorrow for the file permission

If that does not work you can always try to do a full uninstall of the UniFi controller and all dependencies for Java... 
Start off with service unifi.sh stop
rm -rf /usr/local/UniFi rm /usr/local/etc/rc.d/unifi.sh

Run the next 2 commands separately and put a y in to get rid of dependencies on each and press enter... 
pkg remove javavmwrapper
pkg remove  java-zoneinfo

Then go ahead and rerun the fetch link, that should completely fix you, as that will fully remove all of the java stuff, and by running the fetch link reinstall it.

tong2x commented:
_**service unifi.sh stop
rm -rf /usr/local/UniFi
rm /usr/local/etc/rc.d/unifi.sh

pkg remove javavmwrapper
pkg remove java-zoneinfo**_

your really good at this things gnkidwell, that solve the problem. attach also is the log file.

I just have question, the new script does not have a download progress bar unlike the original script, was it removed?

2nd, the install script did not end gracefully, maybe because of the warn message at end? is this normal, well not normal, rather is this ok?
[unifi6028.txt](https://github.com/gozoinks/unifi-pfsense/files/5426625/unifi6028.txt)
![unifiintall](https://user-images.githubusercontent.com/40678733/96943768-a448f380-150b-11eb-9832-ce8f1632ef69.png)

gnkidwell commented:
> _**service unifi.sh stop
> rm -rf /usr/local/UniFi
> rm /usr/local/etc/rc.d/unifi.sh
> 
> pkg remove javavmwrapper
> pkg remove java-zoneinfo**_
> 
> your really good at this things gnkidwell, that solve the problem. attach also is the log file.
> 
> I just have question, the new script does not have a download progress bar unlike the original script, was it removed?
> 
> 2nd, the install script did not end gracefully, maybe because of the warn message at end? is this normal, well not normal, rather is this ok?
> [unifi6028.txt](https://github.com/gozoinks/unifi-pfsense/files/5426625/unifi6028.txt)
> ![unifiintall](https://user-images.githubusercontent.com/40678733/96943768-a448f380-150b-11eb-9832-ce8f1632ef69.png)

So the progress bar only exists with the standard fetch link, when outputting to text you lose the download bar, totally normal behavior. The WARN message is normal as this would be the behavior of a fresh install of the UniFi controller. As far as it ending gracefully, it does, if you did a ctrl-z it would not even show suspended. It just does not return a prompt without a key press... You should be good for a while again... Let me know if you need anything else.

tong2x commented:
hello gnkidwell
it would seem our issue with java is something related to the sensei plugin which is using the elasticsearch included in removal of package. when I run "pkg remove javavmwrapper"
i think they did made updates, 2 or 3 weeks ago.

in another system of mine, I could not "stop" the service as the java error appears, and it does not stop the service.
any way to stop the service forcefully?
or run "rm /usr/local/etc/rc.d/unifi.sh" then reboot, so service wont start and I can make a full clean removable?

gnkidwell commented:
Type top find the process id for the process... Ctrl-c... Kill processidhere

On October 22, 2020 9:30:09 PM CDT, tong2x <notifications@github.com> wrote:
>hello gnkidwell
>it would seem our issue with java is something related to the sensei
>plugin which is using the elasticsearch included in removal of package.
>when I run "pkg remove javavmwrapper"
>i think they did made updates, 2 or 3 weeks ago.
>
>in another system of mine, I could not "stop" the service as the java
>error appears, and it does not stop the service.
>any way to stop the service forcefully?
>or run "rm /usr/local/etc/rc.d/unifi.sh" then reboot, so service wont
>start and I can make a full clean removable?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/gozoinks/unifi-pfsense/pull/211#issuecomment-714868456

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

tong2x commented:
just wan to report that for the other unit, I used "rm /usr/local/etc/rc.d/unifi.sh"
then rebooted and started the install script again for build 28

I did executed pkg remove javavmwrapper, pkg remove java-zoneinfo
before running the script

I did not have to clean uninstall everything, the script basically updated my install and fixed the java issue all together

gnkidwell commented:
@tong2x I know you are generally interested in running the beta controllers. So I am forking off the beta stuff to my repository... If you are interested I currently have 6.0.31 available https://github.com/gnkidwell/unifi-pfsense/tree/beta ... For now under this gozoinks master repository I will only be posting stable releases, since the last pull update here was 2 months ago. Also if you want to submit changes their you can, and I will review them. I likely will be creating a separate folders with different version numbers as well. See you there 👍🏻

tong2x commented:
thanks will take a look at your repo soon and try it out

tong2x commented:
the java issue is back at it most likely is this file.
so uninstalled java again and tried installer, it was successfull
but after install I run opnsense update and it would like to reupdate jdsk8
which is probably the one causing the java error...
![opnsenseJDK8](https://user-images.githubusercontent.com/40678733/97242816-d0c47e80-182f-11eb-8dee-924888bcdf9e.png)
 

btw, why cant I create a issue in your beta? or maybe Im not just seeing the button

gnkidwell commented:
> the java issue is back at it most likely is this file.
> so uninstalled java again and tried installer, it was successfull
> but after install I run opnsense update and it would like to reupdate jdsk8
> which is probably the one causing the java error...
> ![opnsenseJDK8](https://user-images.githubusercontent.com/40678733/97242816-d0c47e80-182f-11eb-8dee-924888bcdf9e.png)
> 
> btw, why cant I create a issue in your beta? or maybe Im not just seeing the button

You cannot create an issue in a fork... But... For now I would not use OPNSense to update packages, especially considering that is the same exact version of Java. I mean you could try to lock openjdk8... PFSense does not even have an option to update packages like openjdk8 from the UI... I know you are running a specific plugin... But I cannot support every plugin, for now try to lock openjdk8..

tong2x commented:
correct, that what I'll do for now skip openjdk8

gnkidwell commented:
I just patched a few things. Found out by the V8 port being down on FreeBSD, that V8 is not necessary, so I went ahead and got rid of Python27 for Python37... I also got rid of line 99, which is currently line 100 of the commit, this line for the base.txz file is completely unnecessary and does not effect the installation of the controller. I added a unlock and lock of the packages that are installed by the install-unifi.sh script... This was done for situations where a user might be running something like OPNSense where updates from OPNSense like to reinstall packages which can effectively break the UniFi controller. This currently is the most stable script yet, of course all depedency errors were also fixed.

If you did install this script prior feel free to follow the directions below to get rid of python27 and v8...
service unifi.sh stop
pkg remove -y python27
pkg remove -y v8
fetch -o - https://git.io/JTD2a | sh -s

tong2x commented:
confirmed 
no more base.txz error in opensenze
Unifi build 28 loaded without any problems or errors

only a warn message at end which is not critical
root@OPNsense:~ # WARN Unable to load properties from '/usr/local/UniFi/data/system.properties' - /usr/local/UniFi/data/system.properties (No such file or directory)

gnkidwell commented:
> confirmed 
> no more base.txz error in opensenze
> Unifi build 28 loaded without any problems or errors
> 
> only a warn message at end which is not critical
> root@OPNsense:~ # WARN Unable to load properties from '/usr/local/UniFi/data/system.properties' - /usr/local/UniFi/data/system.properties (No such file or directory)

There is no way to get rid of that WARN message on a fresh install.. Due to the fact, that system.properties has not been configured until after the controller starts for the first time. At this time I have exhausted all of the options.

tong2x commented:
official 6.0.36 is out

gnkidwell commented:
> official 6.0.36 is out

Thanks I will close this one out, and make a new one for 6.0.36. Also I went ahead and merged your commit to my fork.

gnkidwell commented:
Closed due to new official release 6.0.36 from Ubiquiti. Can be found here in #215
